components/director/internal/domain/tenant/repository.go

```diff
@@ -370,6 +370,45 @@ func (r *pgRepository) GetLowestOwnerForResource(ctx context.Context, resourceTy
 	return dest.TenantID, nil
 }
 
+// GetCustomerIDParentRecursively gets the top parent external ID (customer_id) for a given tenant
+func (r *pgRepository) GetCustomerIDParentRecursively(ctx context.Context, tenantID string) (string, error) {
+	recursiveQuery := `WITH RECURSIVE parents AS
+                   (SELECT t1.id, t1.parent, t1.external_tenant, t1.type
+                    FROM business_tenant_mappings t1
+                    WHERE id = $1
+                    UNION ALL
+                    SELECT t2.id, t2.parent, t2.external_tenant, t2.type
+                    FROM business_tenant_mappings t2
+                             INNER JOIN parents p on p.parent = t2.id)
+			SELECT external_tenant, type FROM parents WHERE parent is null`
+
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	log.C(ctx).Debugf("Executing DB query: %s", recursiveQuery)
+
+	dest := struct {
+		ExternalCustomerTenant string `db:"external_tenant"`
+		Type                   string `db:"type"`
+	}{}
+
+	if err := persist.GetContext(ctx, &dest, recursiveQuery, tenantID); err != nil {
+		return "", persistence.MapSQLError(ctx, err, resource.Tenant, resource.Get, "while getting parent external customer ID for internal tenant: %q", tenantID)
+	}
+
+	if dest.Type != tenant.TypeToStr(tenant.Customer) {
+		return "", nil
```

Comment:
The funtion looks for CustomerID and such is not found. Maybe an error should be returned herer? Otherwise there should be checks for empty string whenever the function is invoked which is errorprone.

Reply:
If no customer tenant is found, we are OK with it - just return empty string and use in the notification payload.